client/wsclient.go

```diff
@@ -97,22 +161,37 @@ func (c *wsClient) AgentDescription() *protobufs.AgentDescription {
 }
 
 func (c *wsClient) SetAgentDescription(descr *protobufs.AgentDescription) error {
+	if c.isStopped.Load() {
+		return errStopping
+	}
```

Comment:
> What is the purpose of this check?
This check disallows client implementations from attempting to send messages after they have called `Stop`.

>What happens if Stop() is called immediately after this check, is there a danger of race?
There is, that's a great observation. We need to inhibit stopping for the duration of these methods to prevent the messages from being lost without an error being returned. I've updated the concurrency control mechanisms to account for this.